tensorflow/compiler/xla/client/lib/math.cc

```diff
@@ -1112,10 +1112,26 @@ XlaOp RoundToEven(XlaOp x) {
 // acos(x) = 2 * atan(sqrt(1 - x^2) / (1 + x)) if x != -1
 //           pi                                if x == -1
 XlaOp Acos(XlaOp x) {
-  return Select(Ne(x, FullLike(x, -1)),
-                ScalarLike(x, 2.0) * Atan2(Sqrt(ScalarLike(x, 1.0) - x * x),
-                                           ScalarLike(x, 1.0) + x),
-                FullLike(x, M_PI));
+  XlaBuilder* b = x.builder();
+  return b->ReportErrorOrReturn([&]() -> StatusOr<XlaOp> {
+    TF_ASSIGN_OR_RETURN(auto shape, b->GetShape(x));
+
+    // complex: acos(x) = -i * log(x + sqrt(-(x+1)*(x-1)))
```

Comment:
The code on line 1127 multiplies the `sqrt` by `-i`. The expression inside the `sqrt` is also a bit different.
Please eliminate the inconsistencies and move the comment up before the method.